### ISSUE TITLE
new name index generator

### ISSUE THREAD

**Issue by rbuels** [GMOD/jbrowse-components#56]
I think generate-names is probably the only major part of the Perl codebase we're going to be wanting to carry into JBrowse 2, so we should probably port it to node for 2.0.

We would keep the Perl version of it in JBrowse 1 of course.

depends on #55

**Comment by rbuels** [comment]
but don't forget about the importance of incremental indexing.

**Comment by bbimber** [comment]
@rbuels This is clearly an old thread, but I wanted to check in: has there been other discussion around this? This feature is important to us, and preventing us from migrating to JB2.

Like I mentioned on jitter a while back we might be able to do this development, if we could do this in a way that works for you and if that helps accelerate progress here. I'm bringing on a dev and would consider this as a first project for them.

In general, i assume the goal is approximate feature parity with JB1, except ported to npm/react. I havent really thought in detail, but one other nice-to-have would be a more federated approach to the index. My sense on JB2 is that there is a lot more support for supplying tracks by URL, meaning there could be one instance of the FASTA, GTF, GFF, VCF, etc. that is consumed by more than one JB2 instance. It would be great if components of that index could be created per-track, and somehow unified. Alternately perhaps at least each track could pre-index its features, and then some kind of aggregator step could take those and merged into one.

Anyway, I'd like to start some discussion around whether porting name indexing and search if viable, and offer our help to make it happen.

**Comment by rbuels** [comment]
It's a longstanding issue, but it's very much alive. The other main issue in this cluster is #1163 

We could definitely use some help with it, since we have some grant deliverables that take priority over this for the next half-year to a year.

I've got your direct email, I'm going to email that, and let's set up a call to plan how we can work together to get this done.

**Comment by rbuels** [comment]
closing in favor of new formulation of this in #1827
